### PR TITLE
Derelict mommis self-destruct if they find their way off the zlevel

### LINKED ISSUE #13187: Derelict MoMMis can teleport to the station when inside a cage
(WEB REPORT BY: archsunder REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)
fcaa64cba57915006c8c3fcdf22ea1e28c73bbbd 

>General description of the issue
BROKE THE CONDITIONING and managed to teleport to the carbon station whilst inside a plasteel cage

>What you expected to happen
"You can't use this teleporter"

>What actually happened
Appeared on carbon station, beeped and pinged at some carbons, then teleported home

>Steps to reproduce if possible

### PR DESCRIPTION
**Pull request by Sprok0**
If someone manages to find a way around all the checks on z-locked mommis, they now gib after leaving their designated zlevel.

Fixes #13187
Fixes being able to take them off their level with shuttles
Fixes future ways around z-level locking(for mommis)

### CODE REVIEW

**Review by PJB3005 — APPROVED** [review]

File: code/modules/mob/living/silicon/mommi/life.dm [hunk 1]
```diff
@@ -263,4 +265,18 @@
...
 
+/mob/living/silicon/robot/mommi/proc/check_locked_zlevel()
+	if(!locked_to_z)
+		return
+
+	var/datum/zLevel/current_zlevel = get_z_level(src)
+	if(!current_zlevel)
+		return
+	if(current_zlevel.z != locked_to_z)
+		to_chat(src, "<span class='userdanger'>Your hardware detects that you have left your intended location. Initiating self-destruct.</span>")
```
Comment (9600bauds) [inline code/modules/mob/living/silicon/mommi/life.dm:276]: Give this a spawn(5 SECONDS) for fun!!!

**Comment by 9600bauds** [comment]
Also, I think this doesn't activate if they're inside a locker, for all THAT'S worth (if you want to spend the rest of your life in a locker)

**Comment by 9600bauds** [comment]
or actually more worryingly it might destroy them the very moment they climb into a locker inside the derelict, honk

**Comment by Sprok0** [comment]
It doesn't, I checked that specifically. get_z_level() uses get_turf() to determine which level something is on.

**Comment by 9600bauds** [comment]
All gucchi then 👌

File: code/modules/mob/living/silicon/mommi/life.dm [hunk 2]
```diff
@@ -263,4 +265,18 @@
...
+		if(mmi) //no sneaking brains away
```
Comment (ghost) [inline code/modules/mob/living/silicon/mommi/life.dm]: You could use dust() which is functionally like gib() except the mmi is qdeled for you so it saves you two lines of copy paste here

Comment (Sprok0) [reply code/modules/mob/living/silicon/mommi/life.dm]: I wanted them to burst into parts, not melt into remains. It's all about the aesthetics

File: code/modules/mob/living/silicon/mommi/life.dm (hunk 1 above)
Comment (9600bauds) [reply code/modules/mob/living/silicon/mommi/life.dm:276]: Done

**Comment by Intigracy** [comment]
I think you should remove all the checks and let mommis leave the z level, but keep the gibbing.

My immersions.

**Comment by clusterfack** [comment]
Why

**Comment by 9600bauds** [comment]
contain it?

**Comment by Kurfursten** [comment]
@9600bauds Since this PR was brought on because MoMMIs snuck off the derelict in cages, I think it's only appropriate to point out that the sound they'll make rattling their cages will serve as a warning to the rest

**Comment by Intigracy** [comment]
Because there's no need for so many checks if it'll just gib them when they get around them to leave.

Let them commit suicide by jumping into space instead of bouncing off the edges of the z level.

**Review by Probe1 — APPROVED** [review]
gj making sure the brain gets deleted.

sorry players, we've wasted enough man hours and bans on this.

**Comment by Probe1** [comment]
Changelog please.  Players should be told.

**Comment by Shadowmech88** [comment]
I'm going to miss seeing players invent new ways to escape.

**Comment by Probe1** [comment]
Me too but lets face it.  We've spent a lot of time.  A lot of _your_ time closing loopholes and issuing bans that could have been spent coding new content and bussing rounds.

**Comment by Sprok0** [comment]
Spawned MoMMIs also now don't have an MMI inside themselves, I've seen that conveyor belt leading into the teleporter trick and there's really no reason for them to have one.

**Comment by Shadowmech88** [comment]
So what comes out if they get gibbed? Just a brain?

**Comment by PJB3005** [comment]
No the point is that they don't drop anything you could use to revive them via becoming an MoMMI.

Probably nothing?

**Comment by Sprok0** [comment]
Nothing, they just pop into robogibs.
If you use the forgotten feature of putting an MMI directly into the mommi spawner, the resulting mommi still has an MMI though.

**Comment by ghost** [comment]
How about hats , @Sprok0 ?

**Comment by Sprok0** [comment]
I didn't touch their hat abilities

**Comment by ghost** [comment]
No no I mean do hats qdel on gib?

**Comment by Sprok0** [comment]
No, the hat remains intact

**Comment by Probe1** [comment]
And all that was left was a cloud of smoke a hat and a faint odor of autism

**Pull request merged**